packages/dev/core/src/Shaders/ShadersInclude/lightFragment.fx

```diff
@@ -6,7 +6,35 @@
         vec4 diffuse{X} = light{X}.vLightDiffuse;
         #define CUSTOM_LIGHT{X}_COLOR // Use to modify light color. Currently only supports diffuse.
 
-        #ifdef PBR
+        #if defined(PBR) && defined(CLUSTLIGHT{X}) && CLUSTLIGHT_BATCH > 0
+            info = computeClusteredLighting(
+                lightDataTexture{X},
+                tileMaskTexture{X},
+                light{X}.vLightData,
+                int(light{X}.vNumLights),
+                viewDirectionW,
+                normalW,
+                vPositionW,
+                surfaceAlbedo,
+                reflectivityOut
+                #ifdef SS_TRANSLUCENCY
+                    , subSurfaceOut
+                #endif
+                #ifdef SPECULARTERM
+                    , AARoughnessFactors.x
+                #endif
+                #ifdef ANISOTROPIC
+                    , anisotropicOut
+                #endif
+                #ifdef SHEEN
+                    , sheenOut
+                #endif
+                #ifdef CLEARCOAT
+                    , clearcoatOut
+                #endif
+            );

```

Comment:
`numLights` must be updated according to the number of lights processed by `computeClusteredLighting` (minus 1, because we have `numLights += 1.0` later on in the file).

Reply:
`numLights` is only used (afaict) to divide `aggShadow` so we'd either have to adjust `aggShadow` accordingly too or just leave it as updating both by 1

Reply:
checked offline